### PR TITLE
executor,table: remove redundant if condition and fix comment typos

### PR DESCRIPTION
**Pull request by exialin**
### What problem does this PR solve? 

Some minor changes to make the code more readable.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8627)

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: table/column.go [hunk 1]
```diff
@@ -286,7 +286,7 @@ func CheckOnce(cols []*Column) error {
 
 // CheckNotNull checks if nil value set to a column with NotNull flag is set.
 func (c *Column) CheckNotNull(data types.Datum) error {
-	if (mysql.HasNotNullFlag(c.Flag) && data.IsNull() || mysql.HasPreventNullInsertFlag(c.Flag)) && data.IsNull() {
```
Comment (zz-jason) [inline table/column.go]: why `data.IsNull()` can be removed?

Comment (exialin) [reply table/column.go]: Because `(a && b || c) && b = (a && b && b) || (c && b) = (a && b) || (c && b) = (a || c) && b`.

**Comment by zhouqiang-cl** [comment]
@XuHuaiyu jenkins-ci-tidb/integration-ddl-test  is old ci which is not in use now.
the instead is idc-jenkins-ci-tidb/integration-ddl-test

File: table/column.go (hunk 1 above)
Comment (zz-jason) [reply table/column.go]: Got it.

**Pull request merged**